examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -0,0 +1,284 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+
+const String butterflyUri =
+    'https://flutter.github.io/assets-for-api-docs/videos/butterfly.mp4';
+
+const String beeUri =
+    'https://flutter.github.io/assets-for-api-docs/videos/bee.mp4';
+
+class LoopingController {
+  final String uri;
+
+  LoopingController(this.uri);
+
+  VideoPlayerController _controller;
+  VideoPlayerController get controller => _controller;
+
+  Future<Null> init() async {
+    _controller = await VideoPlayerController.create(uri);
+    await _controller.setLooping(true);
+    await _controller.play();
+  }
+
+  Future<Null> dispose() async {
+    await _controller.dispose();
+  }
+}
+
+class LoopingVideoCard extends StatelessWidget {
+  final VideoPlayerController controller;
+  final String title;
+  final String subtitle;
+
+  const LoopingVideoCard({Key key, this.controller, this.title, this.subtitle})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
This is a very long build method, I suggest refactoring it a bit. For example, the onTap builds a whole screen, could that be a separate widget (i.e. FullScreenVideo)?

Reply:
Good idea. I split it into a couple of helper methods. Hope it reads nicer.